code/game/machinery/Sleeper.dm

```diff
@@ -229,10 +288,30 @@
 	return
 
 
-/obj/machinery/sleeper/attackby(obj/item/weapon/grab/G as obj, mob/user as mob)
-	if((!( istype(G, /obj/item/weapon/grab)) || !( ismob(G.affecting))))
+/obj/machinery/sleeper/attackby(obj/item/weapon/W as obj, mob/user as mob)
+	if(iscrowbar(W) && occupant)

```

Comment:
You should be including this in crowbarDestroy, not in the attackby, because something could otherwise manually call that code and bypass this.
See how the cloner does it.